mne/decoding/tests/test_transformer.py

```diff
@@ -229,3 +229,32 @@ def test_unsupervised_spatial_filter():
     usf = UnsupervisedSpatialFilter(PCA(4), average=True)
     usf.fit_transform(X)
     assert_raises(ValueError, UnsupervisedSpatialFilter, PCA(4), 2)
+
+
+def test_temporal_filterer():
+    """Test methods of TemporalFilter."""
+    X = np.random.rand(5, 5, 1200)
+
+    # Test init test
+    values = (('10hz', None, 100., 'auto'), (5., '10hz', 100., 'auto'),
+              (10., 20., 5., 'auto'), (None, None, 100., '5hz'))
+    for low, high, sf, ltrans in values:
+        assert_raises(ValueError, TemporalFilter, low, high, sf, ltrans)
+
+    # Add tests for different combinations of l_freq and h_freq
+    for low, high in ((5., 15.), (None, 15.), (5., None)):
+        with warnings.catch_warnings(record=True):
+            filt = TemporalFilter(low, high, sfreq=100.)
+            Xt = filt.fit_transform(X)
+            assert_array_equal(filt.fit_transform(X), Xt)
+        assert_true(X.shape == Xt.shape)
+
+    # Test fit and transform numpy type check
+    with warnings.catch_warnings(record=True):
+        assert_raises(TypeError, filt.transform, [1, 2])
+
+    # Test with 2 dimensional data array
+    X = np.random.rand(10, 5)
```

Comment:
I get the following warning.
```DeprecationWarning: filter_length (265) is longer than the signal (2), distortion is likely. Reduce filter length or filter a longer signal.
  filt = TemporalFilter(l_freq=10., h_freq=15., sfreq=100.)
```
Even if I increase the samples and time trials the warning persists. Also I have taken small samples to make the test fast.